vpr/src/route/router_lookahead_map_utils.cpp

```diff
@@ -73,29 +76,29 @@ util::PQ_Entry_Delay::PQ_Entry_Delay(
         }
 
         VTR_ASSERT(T_linear >= 0.);
-        this->delay_cost += T_linear;
+        this->delay_cost = parent->delay_cost + T_linear;
+    } else {
+        this->delay_cost = 0.f;
     }
 }
 
 util::PQ_Entry_Base_Cost::PQ_Entry_Base_Cost(
     int set_rr_node_ind,
     int switch_ind,
-    float upstream_base_costs,
-    bool starting_node) {
+    const util::PQ_Entry_Base_Cost* parent) {
     this->rr_node_ind = set_rr_node_ind;
 
-    auto& device_ctx = g_vpr_ctx.device();
-    this->base_cost = upstream_base_costs;
-    if (!starting_node) {
-        if (device_ctx.rr_switch_inf[switch_ind].configurable()) {
-            this->base_cost += get_rr_cong_cost(set_rr_node_ind);
-        }
+    if (parent != nullptr) {
+        auto& device_ctx = g_vpr_ctx.device();
+        this->base_cost = parent->base_cost + (device_ctx.rr_switch_inf[switch_ind].configurable() ? get_rr_cong_cost(set_rr_node_ind) : 0);
```

Comment:
Please leave this on multiple lines, no need to compact it and blow up the character count.

Reply:
Okay, I expanded it. I like do assignments at once instead of conditional increments because it makes the equation easier to understand.